implementation_test.go

```diff
@@ -7,17 +7,163 @@ import (
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(2), res)
+	}
+}
+
+func TestCalculatePostfix_Multiplying_Integers(t *testing.T) {
+	res, err := CalculatePostfix("5 18 * 2 *")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(180), res)
+	}
+}
+
+func TestCalculatePostfix_Multiplying_Floats(t *testing.T) {
+	res, err := CalculatePostfix("6.25 5.8 *")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(36.25), res)
+	}
+}
+
+func TestCalculatePostfix_Multiplying_Negatives(t *testing.T) {
+	res, err := CalculatePostfix("-9 11 *")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(-99), res)
+	}
+}
+
+func TestCalculatePostfix_Multiplying_ByZero(t *testing.T) {
+	res, err := CalculatePostfix("13 0 *")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(0), res)
+	}
+}
+
+func TestCalculatePostfix_Dividing_Integers(t *testing.T) {
+	res, err := CalculatePostfix("18 5 / 2 /")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(1.8), res)
+	}
+}
+
+func TestCalculatePostfix_Dividing_Floats(t *testing.T) {
+	res, err := CalculatePostfix("9.5 8.0 /")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(1.1875), res)
+	}
+}
+
+func TestCalculatePostfix_Dividing_Negatives(t *testing.T) {
+	res, err := CalculatePostfix("-45 -15 / -3 /")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(-1), res)
+	}
+}
+
+func TestCalculatePostfix_Dividing_ByZero(t *testing.T) {
+	_, err := CalculatePostfix("13 0 /")
+	assert.Error(t, err, "String with dividing by zero should return error")
+}
+
+func TestCalculatePostfix_Exponentiating_IntegerExponent(t *testing.T) {
+	res, err := CalculatePostfix("6 3 ^")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(216), res)
+	}
+}
+
+func TestCalculatePostfix_Exponentiating_FloatExponent(t *testing.T) {
+	res, err := CalculatePostfix("0.64 0.5 ^")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(0.8), res)
+	}
+}
+
+func TestCalculatePostfix_Exponentiating_NegativeExponent(t *testing.T) {
+	res, err := CalculatePostfix("2 -5 ^")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(0.03125), res)
+	}
+}
+
+func TestCalculatePostfix_Exponentiating_ZeroExponent(t *testing.T) {
+	res, err := CalculatePostfix("14 0 ^")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(1), res)
+	}
+}
+
+func TestCalculatePostfix_ComplexExpression_1(t *testing.T) {
+	res, err := CalculatePostfix("6 10 + 4 - 1 1 2 * + / 1 +")
+	if assert.Nil(t, err) {
+		assert.Equal(t, float64(5), res)
+	}
+}
+
+func TestCalculatePostfix_ComplexExpression_2(t *testing.T) {
+	_, err := CalculatePostfix("3 7 - 6 + 2 2 - /")
+	assert.Error(t, err, "String with dividing by zero should return error")
+}
+
+func TestCalculatePostfix_ComplexExpression_3(t *testing.T) {
+	res, err := CalculatePostfix("5 8 + 3 - 2 ^ 5 2 * /")
 	if assert.Nil(t, err) {
-		assert.Equal(t, "4 2 - 3 * 5 +", res)
+		assert.Equal(t, float64(10), res)
 	}
 }
 
-func ExamplePrefixToPostfix() {
-	res, _ := PrefixToPostfix("+ 2 2")
+func ExampleCalculatePostfix() {
```

Comment:
@lightballer @olegmak02 Do we need to add more code to this function? And should it print result with `fmt.Println(res)` instead of running it like test with `assert`? I know that this code from template but here what we can find in specification:
<img width="776" alt="image" src="https://user-images.githubusercontent.com/35728917/140803391-225500ec-e764-41ae-80c5-726374267bf7.png">

Reply:
maybe